### ISSUE TITLE
Issue with header parsing

### ISSUE THREAD

**Issue by lyrm** [mirage/mrmime#69]
Seed 102. 

There is an issue with the header parsing. From what I found, everything is fine until the call to ```Unstrctrd_parser.unstrctrd``` in ```Mrmime.Field.Decoder.field```. If you peek at the parsed string :
-  before this call : everything is ok, the pointer is just before the header value. 
- after this call : something went wrong and the pointer has moved to the next line. 

However I don't know if the issue comes from the ```Unstrctrd``` function or if it comes from how the header are encoded.

**Comment by lyrm** [comment]
It is actually the expected behavior (I thought this function was similar to a peek function).

**Comment by lyrm** [comment]
For information, the problematic mail looks like that: 
```
Resent-Bcc: 28 May 1348 09:25:45 MDT
Content-Type: multipart/mixed; boundary="ei4DWr+ZGc0="; 
                 hm7en="2G?cB#(L6/\f"

--ei4DWr+ZGc0=
Resent-Date: 25 Mar 504 19:52:20 CDT


--ei4DWr+ZGc0=
Resent-Sender: quoted-printable


--ei4DWr+ZGc0=--
```
And the issue actually comes from  ```Unstrctrd.without_comments``` that returns an error for the second header (the content-type one). I will continue my investigation :monocle_face:.

**Comment by lyrm** [comment]
Ok, so I guess the issue is simply that there is an open parenthesis in the header value and no closing one, which is considered as an not closed comment which results in the error. 

However shouldn't we ignore the parenthesis since it is inside two ```"``` ?

**Comment by dinosaure** [comment]
> However shouldn't we ignore the parenthesis since it is inside two " ?

Yes, `unstrctrd` should be smarter than that. I will try to think about a proper solution.

**Comment by lyrm** [comment]
I'm working on improving the ```without_comments``` function, but that is not as simple as I thought :sweat_smile:.

**Comment by dinosaure** [comment]
> I'm working on improving the without_comments function, but that is not as simple as I thought 😅.

Yeah, the situation is a bit complex and I need to check some details on RFCs, at least, you should be able to fix the fuzzer to not generate parenthesis for such values and I will try to find a good solution.

**Comment by dinosaure** [comment]
Fixed by #70
